core/os/process.cpp

```diff
@@ -0,0 +1,238 @@
+
+	return start();
+}
+
+bool Process::start() {
+	ERR_FAIL_COND_V(process_state != STATE_NOT_RUNNING, false);
+
+	exit_code = 0;
+	exit_status = EXIT_STATUS_NORMAL;
+
+	return _start();
+}
+
+bool Process::can_read_line() const {
+	return next_line_size() >= 0;
+}
+
+void Process::_set_environment_bind(const Dictionary &p_env, bool p_override) {
+	environment.clear();
+
+	List<Variant> keys;
+	p_env.get_key_list(&keys);
+
+	for (List<Variant>::Element *E = keys.front(); E; E = E->next()) {
+		environment.set(E->get(), p_env[E->get()]);
+	}
+
+	if (!p_override) {
+		_get_system_env(environment);
+	}
+}
+
+Dictionary Process::_get_environment_bind() const {
+	Dictionary dict;
+
+	const String *k = NULL;
+	while ((k = environment.next(k))) {
+		dict[k] = environment.get(*k);
+	}
+
+	return dict;
+}
+
+PoolVector<uint8_t> Process::_read_all_bind() {
+	PoolVector<uint8_t> ret;
+	ret.resize(get_available_bytes());
+	PoolVector<uint8_t>::Write w = ret.write();
+	read_all((char *)w.ptr(), ret.size());
+	return ret;
+}
+
+PoolVector<uint8_t> Process::_read_line_bind() {
+	PoolVector<uint8_t> ret;
+	int size = next_line_size();
+	if (size == -1)
+		return ret;
+	ret.resize(size);
+	PoolVector<uint8_t>::Write w = ret.write();
+	read_line((char *)w.ptr(), ret.size());
+	return ret;
+}
+
+int Process::_write_bind(const PoolVector<uint8_t> &p_bytes) {
+	PoolVector<uint8_t>::Read r = p_bytes.read();
+	return write((const char *)r.ptr(), p_bytes.size());
+}
+
+bool Process::_start_bind(const String &p_program, const Vector<String> &p_arguments) {
+	set_program(p_program);
+	set_arguments(p_arguments);
+	return start();
+}
+
+void Process::close_channel(Process::ProcessChannel p_channel) {
+	switch (p_channel) {
+		case CHANNEL_STDOUT: {
+			stdout_chan->close();
+		} break;
+		case CHANNEL_STDERR: {
+			stderr_chan->close();
+		} break;
+		case CHANNEL_STDIN: {
+			stdin_sama->close();
+		} break;
+		default: ERR_FAIL();
+	}
+}
+
+Ref<Process> Process::create_ref() {
+	if (!_create)
+		return NULL;
+	return Ref<Process>(_create());
+}
+
+Process *Process::create() {
+	if (!_create)
+		return NULL;
+	return _create();
+}
+
+void Process::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("get_program"), &Process::get_program);
+	ClassDB::bind_method(D_METHOD("get_arguments"), &Process::_get_arguments_bind);
+
+	ClassDB::bind_method(D_METHOD("set_cwd", "path"), &Process::set_cwd);
+	ClassDB::bind_method(D_METHOD("get_cwd"), &Process::get_cwd);
+
+	ClassDB::bind_method(D_METHOD("get_state"), &Process::get_state);
+	ClassDB::bind_method(D_METHOD("get_exit_status"), &Process::get_exit_status);
+	ClassDB::bind_method(D_METHOD("get_exit_code"), &Process::get_exit_code);
+
+	ClassDB::bind_method(D_METHOD("set_redirect_flags", "flags"), &Process::set_redirect_flags);
+	ClassDB::bind_method(D_METHOD("get_redirect_flags"), &Process::get_redirect_flags);
+
+	ClassDB::bind_method(D_METHOD("set_environent", "env", "override"), &Process::_set_environment_bind, DEFVAL(false));
```

Comment:
I think `set_environent` should be `set_environment`?

`environent` occurs in a few other places as well.